GCEAgent/addresses.go

```diff
@@ -142,3 +147,38 @@ func (a *addresses) set() error {
 
 	return nil
 }
+
+// Filter out forwarded ips based on WSFC (Windows Failover Cluster Settings).
+// If EnableWSFC is set, all ips in the ForwardedIps will be ignore.
+// If WSFCAddresses is set, only ips in the lisft will be filtered out.
+// If both of attributes are set, logic will respect WSFCAddresses
+func (a *addresses) applyWSFCFilter() {
+	var wsfcAddrs []string
+	for _, wsfcAddr := range strings.Split(a.newMetadata.Instance.Attributes.WSFCAddresses, ",") {
+		if len(wsfcAddr) == 0 {
+			continue
+		} else if net.ParseIP(wsfcAddr) == nil {
+			logger.Errorln("ip address for wsfc is not in valid form", wsfcAddr)
+		} else {
+			wsfcAddrs = append(wsfcAddrs, wsfcAddr)
+		}
+	}
+
+	if len(wsfcAddrs) != 0 {
+		interfaces := a.newMetadata.Instance.NetworkInterfaces
+		for idx := range interfaces {
+			filteredList := interfaces[idx].ForwardedIps[:0]
```

Comment:
this is what I get from here to avoid allocating memory. Let me know whether we can use it or prefer more readable way.
https://github.com/golang/go/wiki/SliceTricks#filtering-without-allocating